src/shogun/labels/MulticlassLabelsEncoder.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+#ifndef _MulticlassLabelsEncoder__H__
+#define _MulticlassLabelsEncoder__H__
+
+#include <memory>
+#include <shogun/base/SGObject.h>
+#include <shogun/labels/DenseLabels.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/lib/SGVector.h>
+
+namespace shogun
+{
+
+	class MulticlassLabelsEncoder : public LabelEncoder
+	{
+	public:
+		MulticlassLabelsEncoder() = default;
+
+		~MulticlassLabelsEncoder() = default;
+
+		/** Fit label encoder
+		 *
+		 * @param Target values.
+		 * @return SGVector which contains unique labels.
+		 */
+		SGVector<float64_t> fit(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			if (!can_convert_float_to_int(result_vector))
+			{	
+				std::set<float64_t> s(result_vector.begin(), result_vector.end());
+				io::warn(
+				    "({}) have been converted to (0...{})", 
+						fmt::join(s, ", "),
+							 result_vector.vlen - 1);
+			}
+			return fit_impl(result_vector);
+		}
+		/** Transform labels to normalized encoding.
+		 *
+		 * @param  Target values to be transformed.
+		 * @return Labels transformed to be normalized.
+		 */
+		std::shared_ptr<Labels>
+		transform(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			return std::make_shared<MulticlassLabels>(
+			    transform_impl(result_vector));
+		}
+		/** Transform labels back to original encoding.
+		 *
+		 * @param normailzed encoding labels
+		 * @return original encoding labels
+		 */
+		std::shared_ptr<Labels>
+		inverse_transform(const std::shared_ptr<Labels>& labs) override
```

Comment:
don't we also need a check valid here? Something that ensures that the labels are contiguous? [0,1,2,3,4, ... ] no gaps.
I wrote some code for this in the links I posted. Either re-use or remove my code :)

Reply:
I am thinking whether we need a check valid here, as inverse_transform is to map from internal encoding to origin encoding. for example, {100, 100, 200, 300} -> {0, 0, 1, 2}, {0, 0, 1, 2} are transformed by internal encoding, but it is not continuous

Reply:
ah you are right of course :)